### PR TITLE
bootlist : add support for any device name as input

### PR DESCRIPTION
**Pull request by abdhaleegit**
In a contineous test environement setups where device names are not persistent accross os install, having a unique key like mac addr as yaml input which does not change across reboots, dlpar or os install with different linux flavours, also the legacy interface name and disk names as input works as is.

### CODE REVIEW

**Comment by abdhaleegit** [comment]
```
disks: "/dev/mapper/mpatha"
JOB ID     : 4f06550176a019d8f854825ad844f3f1b52273b7
JOB LOG    : /root/sandbox/avocado-fvt-wrapper/results/job-2023-06-21T09.32-4f06550/job.log
 (1/3) bootlist_test.py:BootlisTest.test_normal_mode;run-3f8c: STARTED
 (1/3) bootlist_test.py:BootlisTest.test_normal_mode;run-3f8c: PASS (0.33 s)
 (2/3) bootlist_test.py:BootlisTest.test_service_mode;run-3f8c: STARTED
 (2/3) bootlist_test.py:BootlisTest.test_service_mode;run-3f8c: PASS (0.33 s)
 (3/3) bootlist_test.py:BootlisTest.test_both_mode;run-3f8c: STARTED
 (3/3) bootlist_test.py:BootlisTest.test_both_mode;run-3f8c: PASS (0.53 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/sandbox/avocado-fvt-wrapper/results/job-2023-06-21T09.32-4f06550/results.html
JOB TIME   : 3.14 s


host_interfaces: "02:5d:c7:45:86:03 02:5d:c7:45:86:04"

[root@ltc common]# avocado run --max-parallel-tasks=1 bootlist_test.py --mux-yaml bootlist_test.py.data/bootlist_test_network.yaml 
JOB ID     : 100f21a7aa50fb0d655d20bcdb78056c6efb8ec8
JOB LOG    : /root/sandbox/avocado-fvt-wrapper/results/job-2023-06-21T09.32-100f21a/job.log
 (1/3) bootlist_test.py:BootlisTest.test_normal_mode;run-f4a2: STARTED
 (1/3) bootlist_test.py:BootlisTest.test_normal_mode;run-f4a2: PASS (0.55 s)
 (2/3) bootlist_test.py:BootlisTest.test_service_mode;run-f4a2: STARTED
 (2/3) bootlist_test.py:BootlisTest.test_service_mode;run-f4a2: PASS (0.55 s)
 (3/3) bootlist_test.py:BootlisTest.test_both_mode;run-f4a2: STARTED
 (3/3) bootlist_test.py:BootlisTest.test_both_mode;run-f4a2: PASS (0.76 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/sandbox/avocado-fvt-wrapper/results/job-2023-06-21T09.32-100f21a/results.html
JOB TIME   : 3.84 s


disks: "/dev/disk/by-id/dm-uuid-mpath-360050768108001b3a8000000000000c4"

[root@lt common]# avocado run --max-parallel-tasks=1 bootlist_test.py --mux-yaml bootlist_test.py.data/bootlist_test_disk.yaml
JOB ID     : df29629d39602505d456386dac210ea09d161edd
JOB LOG    : /root/sandbox/avocado-fvt-wrapper/results/job-2023-06-21T09.34-df29629/job.log
 (1/3) bootlist_test.py:BootlisTest.test_normal_mode;run-5749: STARTED
 (1/3) bootlist_test.py:BootlisTest.test_normal_mode;run-5749: PASS (0.33 s)
 (2/3) bootlist_test.py:BootlisTest.test_service_mode;run-5749: STARTED
 (2/3) bootlist_test.py:BootlisTest.test_service_mode;run-5749: PASS (0.33 s)
 (3/3) bootlist_test.py:BootlisTest.test_both_mode;run-5749: STARTED
 (3/3) bootlist_test.py:BootlisTest.test_both_mode;run-5749: PASS (0.53 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/sandbox/avocado-fvt-wrapper/results/job-2023-06-21T09.34-df29629/results.html
JOB TIME   : 3.18 s




disks: "/dev/disk/by-id/dm-uuid-mpath-360050768108001b3a8000000000000c4 /dev/disk/by-id/dm-uuid-mpath-360050768108001b3a8000000000000c4"


[root@ltc common]# avocado run --max-parallel-tasks=1 bootlist_test.py --mux-yaml bootlist_test.py.data/bootlist_test_disk.yaml
JOB ID     : c5791c8fc3fa1c3f210c3e57db8c51d7ce4a2acc
JOB LOG    : /root/sandbox/avocado-fvt-wrapper/results/job-2023-06-21T09.35-c5791c8/job.log

 (1/3) bootlist_test.py:BootlisTest.test_normal_mode;run-cb34: STARTED
 (1/3) bootlist_test.py:BootlisTest.test_normal_mode;run-cb34: PASS (0.46 s)
 (2/3) bootlist_test.py:BootlisTest.test_service_mode;run-cb34: STARTED
 (2/3) bootlist_test.py:BootlisTest.test_service_mode;run-cb34: PASS (0.46 s)
 (3/3) bootlist_test.py:BootlisTest.test_both_mode;run-cb34: STARTED
 (3/3) bootlist_test.py:BootlisTest.test_both_mode;run-cb34: PASS (0.70 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/sandbox/avocado-fvt-wrapper/results/job-2023-06-21T09.35-c5791c8/results.html
JOB TIME   : 3.62 s
```

**Comment by Naresh-ibm** [comment]
rest looks good to me.

File: io/common/bootlist_test.py [hunk 1]
```diff
@@ -35,21 +38,31 @@ def setUp(self):
...
-        elif self.disk_names is not None:
-            if not self.disk_names:
-                self.cancel("user should specify disk name")
-            self.names = self.disk_names
+        self.disk_names = []
+        self.host_interfaces = []
+        local = LocalHost()
+        interfaces = os.listdir('/sys/class/net')
+        disk_names = self.params.get("disks", default=None)
+        devices = self.params.get("host_interfaces", default=None)
```
[the remark this answers is not in the document]
Comment (abdhaleegit) [reply io/common/bootlist_test.py]: Done

**Review by vaishnavibhat — CHANGES_REQUESTED** [review]
Tried to run the test with a faulty mac address in the yaml file, test reported the below error : 
]# avocado run bootlist.py -m bootlist_test_network.yaml --max-parallel-tasks=1
JOB ID     : f67de2f6080d44d2fb101afc7b6bed7fac2fa2fa
JOB LOG    : /root/new/avocado-fvt-wrapper/results/job-2023-07-27T07.40-f67de2f/job.log
Task exception was never retrieved
future: <Task finished coro=<StatusServer.serve_forever() done, defined at /usr/local/lib/python3.6/site-packages/avocado_framework-100.1-py3.6.egg/avocado/core/status/server.py:40> exception=AttributeError("'Server' object has no attribute 'serve_forever'",)>
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/avocado_framework-100.1-py3.6.egg/avocado/core/status/server.py", line 44, in serve_forever
    await self._server_task.serve_forever()
AttributeError: 'Server' object has no attribute 'serve_forever'
 (1/3) bootlist.py:BootlisTest.test_normal_mode;run-5409: STARTED
 (1/3) bootlist.py:BootlisTest.test_normal_mode;run-5409: ERROR: 'NoneType' object has no attribute 'name' (0.03 s)
 (2/3) bootlist.py:BootlisTest.test_service_mode;run-5409: STARTED
 (2/3) bootlist.py:BootlisTest.test_service_mode;run-5409: ERROR: 'NoneType' object has no attribute 'name' (0.03 s)
 (3/3) bootlist.py:BootlisTest.test_both_mode;run-5409: STARTED
 (3/3) bootlist.py:BootlisTest.test_both_mode;run-5409: ERROR: 'NoneType' object has no attribute 'name' (0.03 s)
RESULTS    : PASS 0 | ERROR 3 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/new/avocado-fvt-wrapper/results/job-2023-07-27T07.40-f67de2f/results.html
JOB TIME   : 3.67 s

Test summary:
bootlist.py:BootlisTest.test_both_mode;run-5409: ERROR
bootlist.py:BootlisTest.test_normal_mode;run-5409: ERROR
bootlist.py:BootlisTest.test_service_mode;run-5409: ERROR

Can this ERROR be handled as CANCEL test ?

**Comment by abdhaleegit** [comment]
> Tried to run the test with a faulty mac address in the yaml file, test reported the below error : ]# avocado run bootlist.py -m bootlist_test_network.yaml --max-parallel-tasks=1 JOB ID : f67de2f6080d44d2fb101afc7b6bed7fac2fa2fa JOB LOG : /root/new/avocado-fvt-wrapper/results/job-2023-07-27T07.40-f67de2f/job.log Task exception was never retrieved future: <Task finished coro=<StatusServer.serve_forever() done, defined at /usr/local/lib/python3.6/site-packages/avocado_framework-100.1-py3.6.egg/avocado/core/status/server.py:40> exception=AttributeError("'Server' object has no attribute 'serve_forever'",)> Traceback (most recent call last): File "/usr/local/lib/python3.6/site-packages/avocado_framework-100.1-py3.6.egg/avocado/core/status/server.py", line 44, in serve_forever await self._server_task.serve_forever() AttributeError: 'Server' object has no attribute 'serve_forever' (1/3) bootlist.py:BootlisTest.test_normal_mode;run-5409: STARTED (1/3) bootlist.py:BootlisTest.test_normal_mode;run-5409: ERROR: 'NoneType' object has no attribute 'name' (0.03 s) (2/3) bootlist.py:BootlisTest.test_service_mode;run-5409: STARTED (2/3) bootlist.py:BootlisTest.test_service_mode;run-5409: ERROR: 'NoneType' object has no attribute 'name' (0.03 s) (3/3) bootlist.py:BootlisTest.test_both_mode;run-5409: STARTED (3/3) bootlist.py:BootlisTest.test_both_mode;run-5409: ERROR: 'NoneType' object has no attribute 'name' (0.03 s) RESULTS : PASS 0 | ERROR 3 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0 JOB HTML : /root/new/avocado-fvt-wrapper/results/job-2023-07-27T07.40-f67de2f/results.html JOB TIME : 3.67 s
> 
> Test summary: bootlist.py:BootlisTest.test_both_mode;run-5409: ERROR bootlist.py:BootlisTest.test_normal_mode;run-5409: ERROR bootlist.py:BootlisTest.test_service_mode;run-5409: ERROR
> 
> Can this ERROR be handled as CANCEL test ?

Thanks for reporting, I have taken care with latest changes, so now the test cancels if the mac is not in the current list

**Comment by abdhaleegit** [comment]
with interface name

```
 avocado run --max-parallel-tasks=1 bootlist_test.py -m bootlist_test.py.data/bootlist_test_network.yaml 
JOB ID     : b1f7bb92541fc625b7ac19d1f9c15711d3e7b957
JOB LOG    : /home/ioci/logs/io_network/results/job-2023-08-28T06.13-b1f7bb9/job.log
 (1/3) bootlist_test.py:BootlisTest.test_normal_mode;run-0fa9: STARTED
 (1/3) bootlist_test.py:BootlisTest.test_normal_mode;run-0fa9: PASS (0.51 s)
 (2/3) bootlist_test.py:BootlisTest.test_service_mode;run-0fa9: STARTED
 (2/3) bootlist_test.py:BootlisTest.test_service_mode;run-0fa9: PASS (0.52 s)
 (3/3) bootlist_test.py:BootlisTest.test_both_mode;run-0fa9: STARTED
 (3/3) bootlist_test.py:BootlisTest.test_both_mode;run-0fa9: PASS (0.70 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/ioci/logs/io_network/results/job-2023-08-28T06.13-b1f7bb9/results.html
JOB TIME   : 8.54 s
```

with mac devices
```
avocado run --max-parallel-tasks=1 bootlist_test.py -m bootlist_test.py.data/bootlist_test_network.yaml 
JOB ID     : 35c2a24de6fb937932f54c573552877afde1d290
JOB LOG    : /home/ioci/logs/io_network/results/job-2023-08-28T06.11-35c2a24/job.log
 (1/3) bootlist_test.py:BootlisTest.test_normal_mode;run-f4a2: STARTED
 (1/3) bootlist_test.py:BootlisTest.test_normal_mode;run-f4a2: PASS (0.86 s)
 (2/3) bootlist_test.py:BootlisTest.test_service_mode;run-f4a2: STARTED
 (2/3) bootlist_test.py:BootlisTest.test_service_mode;run-f4a2: PASS (0.53 s)
 (3/3) bootlist_test.py:BootlisTest.test_both_mode;run-f4a2: STARTED
 (3/3) bootlist_test.py:BootlisTest.test_both_mode;run-f4a2: PASS (0.71 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/ioci/logs/io_network/results/job-2023-08-28T06.11-35c2a24/results.html
JOB TIME   : 8.95 s
```

with wrong mac address
```
avocado run --max-parallel-tasks=1 bootlist_test.py -m bootlist_test.py.data/bootlist_test_network.yaml 
JOB ID     : 4e9346fe5259a1f8b17089683bc69bfb0565b244
JOB LOG    : /home/ioci/logs/io_network/results/job-2023-08-28T06.15-4e9346f/job.log
 (1/3) bootlist_test.py:BootlisTest.test_normal_mode;run-0710: STARTED
 (1/3) bootlist_test.py:BootlisTest.test_normal_mode;run-0710: CANCEL: Please check the network device (0.01 s)
 (2/3) bootlist_test.py:BootlisTest.test_service_mode;run-0710: STARTED
 (2/3) bootlist_test.py:BootlisTest.test_service_mode;run-0710: CANCEL: Please check the network device (0.01 s)
 (3/3) bootlist_test.py:BootlisTest.test_both_mode;run-0710: STARTED
 (3/3) bootlist_test.py:BootlisTest.test_both_mode;run-0710: CANCEL: Please check the network device (0.01 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 3
JOB HTML   : /home/ioci/logs/io_network/results/job-2023-08-28T06.15-4e9346f/results.html
JOB TIME   : 5.45 s

```

**Comment by abdhaleegit** [comment]
waiting for utils to merge

File: io/common/bootlist_test.py [hunk 2]
```diff
@@ -35,21 +38,31 @@ def setUp(self):
...
+        elif disks:
+            for dev in disks.split():
+                dev_path = disk.get_absolute_disk_path(dev)
+                dev_base = os.path.basename(os.path.realpath(dev_path))
+                if 'dm' in dev_base:
+                    dev_base = multipath.get_mpath_from_dm(dev_base)
+                self.disk_names.append(dev_base)
+            self.names = ' '.join(self.disk_names)
+        else:
+            self.cancel("user should specify a boot device name")
```
Comment (vaishnavibhat) [inline io/common/bootlist_test.py:65]: New changes looks good.

LGTM

**Review by PraveenPenguin — APPROVED** [review]

**Pull request merged**